tools/clang/lib/SPIRV/SPIRVEmitter.cpp

```diff
@@ -2150,42 +2150,51 @@ uint32_t SPIRVEmitter::processTextureGatherRGBACmpRGBA(
   // * SamplerState s, float2 location, int2 offset0, int2 offset1,
   //   int offset2, int2 offset3
   //
-  // An additional out uint status parameter can appear in both of the above,
-  // which we does not support yet.
+  // An additional 'out uint status' parameter can appear in both of the above.
   //
   // Parameters for .GatherCmp{Red|Green|Blue|Alpha}() are one of the following
   // two sets:
-  // * SamplerState s, float2 location, int2 offset
-  // * SamplerState s, float2 location, int2 offset0, int2 offset1,
-  //   int offset2, int2 offset3
+  // * SamplerState s, float2 location, float compare_value, int2 offset
+  // * SamplerState s, float2 location, float compare_value, int2 offset1,
+  //   int2 offset2, int2 offset3, int2 offset4
+  //
+  // An additional 'out uint status' parameter can appear in both of the above.
   //
-  // An additional out uint status parameter can appear in both of the above,
-  // which we does not support yet.
+  // TextureCube's signature is somewhat different from the rest.
+  // Parameters for .Gather{Red|Green|Blue|Alpha}() for TextureCube are:
+  // * SamplerState s, float2 location, out uint status
+  // Parameters for .GatherCmp{Red|Green|Blue|Alpha}() for TextureCube are:
+  // * SamplerState s, float2 location, float compare_value, out uint status
   //
   // Return type is always a 4-component vector.
   const FunctionDecl *callee = expr->getDirectCallee();
   const auto numArgs = expr->getNumArgs();
+  const auto *imageExpr = expr->getImplicitObjectArgument();
+  const QualType imageType = imageExpr->getType();
+  const auto imageTypeId = typeTranslator.translateType(imageType);
+  const auto retTypeId = typeTranslator.translateType(callee->getReturnType());
 
-  if (numArgs != 3 + isCmp && numArgs != 6 + isCmp) {
-    emitError("unsupported '%0' method call with status parameter",
-              expr->getExprLoc())
-        << callee->getName() << expr->getSourceRange();
-    return 0;
-  }
+  // If the last arg is an unsigned integer, it must be the status.
+  const bool hasStatusArg =
+      expr->getArg(numArgs - 1)->getType()->isUnsignedIntegerType();
```

Comment:
Nit: `expr->arg_end()->`

Reply:
actually `expr->arg_end()` is the end iterator which cannot be dereferenced.
sadly there is no `rbegin()` method either
`expr->arguments()` also doesn't have an `rbegin()` method.
```cpp
const auto iter = --(expr->arguments().end());
```
and
```cpp
const auto iter = --(expr->arg_end());
```
will also not work because the iterator class only defines the `++` operator and does not define `--` operator

Reply:
Ah, right! Sorry, my bad.